compiler/rustc_error_codes/src/error_codes/E0774.md

```diff
@@ -0,0 +1,24 @@
+`derive` was applied on something which is not a struct, a union or an enum.
+
+Erroneous code example:
+
+```compile_fail,E0773
+trait Foo {
+    #[derive(Clone)] // error!
+    type Bar;
+}
+```
+
+As said above, the `derive` attribute is only allowed on structs, unions or
+enums:
+
+```
+#[derive(Clone)] // ok!
+struct Bar {
+    field: u32,
+}
+```
+
+You can find more information about `derive` in the [Rust Book].
+
+[Rust Book]: https://doc.rust-lang.org/1.30.0/book/second-edition/appendix-03-derivable-traits.html
```

Comment:
Do we need to link to `1.30.0` which is quite old?

Reply:
Try this page with current stable (it's super weird though).

Reply:
> https://doc.rust-lang.org/1.30.0/book/second-edition/appendix-03-derivable-traits.html

Weird, the local version of rust book does show it. Why doesn't the online version does?

file:///home/user/.rustup/toolchains/stable-x86_64-unknown-linux-gnu/share/doc/rust/html/book/appendix-03-derivable-traits.html

Reply:
Works fine on stable: https://doc.rust-lang.org/book/appendix-03-derivable-traits.html